docs/3_fingerprinting.md

```diff
@@ -0,0 +1,246 @@
+# 🔎 Fingerprinting
+
+Fingerprinting is the process of creating uniquely identifyable data about something arbitrarily large. In the context of ReVanced, fingerprinting is essential to be able to find classes, methods and fields without knowing their original names or certain other attributes, which would be used to identify them under normal circumstances.
+
+## ⛳️ Example fingerprint
+
+This page works with the following fingerprint as an example:
+
+```kt
+
+package app.revanced.patches.ads.fingerprints
+
+// Imports
+
+object LoadAdsFingerprint : MethodFingerprint(
+    returnType = "Z",
+    access = AccessFlags.PUBLIC or AccessFlags.FINAL,
```

Comment:
What if method has only one access modifier like:
`public void onAdLoaded()`

Reply:
Then it would just be:
```suggestion
    access = AccessFlags.PUBLIC,
```

Reply:
Got an error: 
`Type mismatch: inferred type is AccessFlags but Int? was expected`

Reply:
Ah, my bad. If I recall correctly, it should instead be:
```suggestion
    access = AccessFlags.PUBLIC.value,
```

Reply:
My example is not wrong:
![image](https://user-images.githubusercontent.com/107796137/201528358-fda4c6ad-e668-4c1b-9864-455d0f6eae52.png)

Reply:
Interesting. I'm not sure then why the compiler places `final` at some methods but not others. Haven't really looked into it.

Reply:
I wanna run `execute()` function for all occurrences of fingerprint method, but it stops after resolving first fingerprint occurrence. How do I accomplish that?

Reply:
What is the problem? Your method has no final access modifier and is only public.

Reply:
> I wanna run `execute()` function for all occurrences of fingerprint method, but it stops after resolving first fingerprint occurrence. How do I accomplish that?

Fingerprints resolve to a single method by design which also happens to be in favour of speed. If you want to change its design, please do so in the patcher repository. Otherwise you have full access over all classes if you need to do something less conventional such as finding all methods with your own fingerprint.

Reply:
ok, got it

Reply:
> What is the problem? Your method has no final access modifier and is only public.

Well it is overloaded, I've to modify all occurences.

Reply:
I do not know the full context, can you please send a screenshot of the method you want to create a fingerprint for?

Reply:
I was creating a patch that would work similarly like Lucky Patcher to remove ads from apps. We have to use ads calling functions at multiple places:
![image](https://user-images.githubusercontent.com/107796137/201537200-f1770d69-875f-45d3-bce9-fe51e7e44f7e.png)

Reply:
It can be done like this, but you can also loop through all classes/methods yourself through context and then patch the methods that way.

Reply:
> It can be done like this, but you can also loop through all classes/methods yourself through context and then patch the methods that way.

how?

Reply:
Via ˋcontext.classesˋ. They are immutable so if you need to modify a class use ˋcontext.proxy(classDef)ˋ.

Reply:
It just adds lots of `return-void` in the first function:
![image](https://user-images.githubusercontent.com/107796137/201735122-416362d0-0d40-4025-aade-46eaa844868b.png)

Reply:
I think this should be moved to Discord #development or its own issue.